ui/media/js/searchable-models.js

```diff
@@ -0,0 +1,586 @@
+            this.populateModels()
+        }, this))
+    }
+
+    saveCurrentSelection(elem, value, path) {
+        this.currentSelection.elem = elem
+        this.currentSelection.value = value
+        this.currentSelection.path = path
+        this.modelFilter.dataset.path = path
+        this.modelFilter.value = value
+        this.modelFilter.dispatchEvent(new Event('change'))
+    }
+    
+    processClick(e) {
+        e.preventDefault()
+        if (e.srcElement.classList.contains('model-file')) {
+            this.saveCurrentSelection(e.srcElement, e.srcElement.innerText, e.srcElement.dataset.path)
+            this.hideModelList()
+            this.modelFilter.focus()
+            this.modelFilter.select()
+        }
+    }
+    
+    findPreviousSibling(elem, previous = true) {
+        let sibling = previous ? elem.previousElementSibling : elem
+        let lastSibling = elem
+        
+        while (sibling && sibling.classList.contains('model-file')) {
+            if (sibling.style.display == 'list-item') return sibling
+            lastSibling = sibling
+            sibling = sibling.previousElementSibling
+        }
+        
+        // no more siblings, look for previous parent if any
+        if (sibling && sibling.classList.contains('model-folder')) {
+            return this.findPreviousSibling(sibling.firstElementChild.lastElementChild, false)
+        }
+        else if (lastSibling.parentElement.parentElement && lastSibling.parentElement.parentElement.previousElementSibling && lastSibling.parentElement.parentElement.previousElementSibling.firstElementChild && lastSibling.parentElement.parentElement.previousElementSibling.firstElementChild.lastElementChild) {
+            return this.findPreviousSibling(lastSibling.parentElement.parentElement.previousElementSibling.firstElementChild.lastElementChild, false)
+        }
+        else if (lastSibling.parentElement.parentElement.previousElementSibling) {
+            return this.findPreviousSibling(lastSibling.parentElement.parentElement.previousElementSibling, false)
+        }
+    }
+    
+    findNextSibling(elem, next = true) {
+        let sibling = next ? elem.nextElementSibling : elem
+        let lastSibling = elem
+    
+        while (sibling && sibling.classList.contains('model-file')) {
+            if (sibling.style.display == 'list-item') return sibling
+            lastSibling = sibling
+            sibling = sibling.nextElementSibling
+        }
+        
+        // no more siblings, look for next parent if any
+        if (lastSibling.nextElementSibling) {
+            return this.findNextSibling(lastSibling.nextElementSibling.firstElementChild.firstElementChild, false)
+        }
+        else if (lastSibling.parentElement.parentElement.nextElementSibling && lastSibling.parentElement.parentElement.nextElementSibling.firstElementChild && lastSibling.parentElement.parentElement.nextElementSibling.firstElementChild.firstElementChild) {
```

Comment:
Any way we can avoid this? :)

Reply:
Rewrote the findSibling() functions for improved readability and with proper comments. Hope you like it. :)